src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +85,14 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        Predicate<Person> predicate = person -> person.getSex() == Person.Sex.WOMAN
+                && person.getAge() >= femaleAge;
+        return peopleList.stream()
+                .filter(predicate)
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
```

Comment:
here you can use only flatMap and save life of one line) 
```suggestion
                .flatMap(person -> person.getCats().stream())
```